### PR TITLE
TESTENG-1324 - All update strategies should return updated object from the API.

### PR DESCRIPTION
**Pull request by jonatanjakoniuk**
Complete the following tasks before merging in your PR:
- [x] If this is a user facing change, update [version.txt](https://github.com/relativitydev/relativity.testing.framework.api/blob/master/version.txt) and [CHANGELOG.md](https://github.com/relativitydev/relativity.testing.framework.api/blob/master/CHANGELOG.md)

### CODE REVIEW

**Comment by jonatanjakoniuk** [comment]
User creation noticed behavior changes
------------------------------------------------------------------------------
Field Name = value						Before PR			After PR
------------------------------------------------------------------------------
Name = null							null				real value
FullName = null						null				real value
Password = real value					real value			null
BetaUser = false						FALSE			TRUE
AdvancedSearchPublicByDefault = true		TRUE			FALSE
NativeViewerCacheAhead = true			TRUE			FALSE
DocumentSkip = enabled					Enabled			Disabled
EnforceViewerCompatibility = true			TRUE			FALSE
ChangeSettings = false					FALSE			TRUE
ChangePassword = false					FALSE			TRUE
MaximumPasswordAge= 90				90				365
ChangePasswordNextLogin=true			TRUE			FALSE
DataFocus=5							5				1
SkipDefaultPreference=Skip				Skip				Normal
EmailPreference=ErrorOnly				ErrorOnly			Default
Client.Number= real value				real value			null


Matter creation noticed behavior changes
------------------------------------------------------------------------------
Field Name = value						Before PR			After PR
------------------------------------------------------------------------------
CreatedOn = null						null				real value
CreatedBy = null						null				real value
LastModifiedOn = null					null				real value
Actions = null							null				empty list
Client.Number = real value				real value			null

**Comment by jonatanjakoniuk** [comment]
User creation noticed behavior changes
| Field Name = value | Before PR | After PR
| --------------------------------| ---------------------| -------------------------| 
| Name = null| null| real value| 
| FullName = null| null| real value| 
| Password = real value| real value| null| 
| BetaUser = false| FALSE| TRUE| 
| AdvancedSearchPublicByDefault = true| TRUE| FALSE| 
| NativeViewerCacheAhead = true| TRUE| FALSE| 
| DocumentSkip = enabled| Enabled| Disabled| 
| EnforceViewerCompatibility = true| TRUE| FALSE| 
| ChangeSettings = false| FALSE| TRUE| 
| ChangePassword = false| FALSE| TRUE| 
| MaximumPasswordAge= 90| 90| 365| 
| ChangePasswordNextLogin=true| TRUE| FALSE| 
| DataFocus=5| 5| 1| 
| SkipDefaultPreference=Skip| Skip| Normal| 
| EmailPreference=ErrorOnly| ErrorOnly| Default| 
| Client.Number= real value| real value| null| 


Matter creation noticed behavior changes
| Field Name = value| Before PR| After PR| 
| ----------------------------------------| ---------------| -----------------------| 
| CreatedOn = null| null| real value| 
| CreatedBy = null| null| real value| 
| LastModifiedOn = null| null| real value| 
| Actions = null| null| empty list| 
| Client.Number = real value| real value| null|

**Comment by jonatanjakoniuk** [comment]
@paulletang I did change target of this PR to `NextMajorRelease` (freshly created base on master) so there we can accumulate few more Breaking changes.

**Comment by jonatanjakoniuk** [comment]
@paulletang high amount of differences for user creation, suggest that we can have some issue in user create strategy (properties are not map correctly or API doesn't allow to pass some of properties)

**Review by paulletang — APPROVED** [review]

File: source/Relativity.Testing.Framework.Api/Services/IAgentService.cs [hunk 1]
```diff
@@ -155,13 +155,14 @@ public interface IAgentService
 		/// Updates the specified Agent.
 		/// </summary>
 		/// <param name="entity">The entity to update.</param>
+		/// <returns>The updated <see href="https://probable-happiness-2926a3e8.pages.github.io/api/Relativity.Testing.Framework.Models.Agent.html">agent</see> entity.</returns>
 		/// <example>
 		/// <code>
 		/// var agentToUpdate = _agentService.Get(agentArtifactId);
 		/// agentToUpdate.Enabled = false;
 		/// _agentService.Update(agentToUpdate);
```
Comment (klaudiaknafel) [inline source/Relativity.Testing.Framework.Api/Services/IAgentService.cs:163]: shouldn't be examples updated?

**Review by klaudiaknafel — APPROVED** [review]

**Pull request merged**